src/block_disagg/block_disagg_read.c

```diff
@@ -79,11 +79,6 @@ __block_disagg_read_multiple(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *block_di
 
     retry = 0;
 
-    /*
-     * Disaggregated storage only supports up to a fixed number of items. We shouldn't ask for more.
-     */
-    WT_ASSERT(session, *results_count <= WT_DELTA_LIMIT);
```

Comment:
Shouldn't we keep this? The limit is still in place.

Reply:
It's a duplicate assert.

Reply:
Oh I see. In that case I'd prefer to keep this earlier one and remove the later one - the earlier we fail, the better.

Reply:
The earlier one is wrong. It's always asserting on the value before we return from the api.

Reply:
Ah, I see. Thanks